drivers/include/cc1xxx_common.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *               2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_cc1xxx CC1100/CC1100e/CC1101/CC1200 common code
+ * @ingroup     drivers_netdev
+ * @ingroup     net_gnrc_netif
+ * @{
+ *
+ * @file
+ * @brief   CC110x/CC1200 adaption for @ref net_gnrc_netif
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ * @author  Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author  Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * Supported Transceivers
+ * ======================
+ *
+ * This adaption layer is written to be used by CC110x and CC1200 transceivers,
+ * but any transceiver using that should transfer 6LoWPAN and uses layer 2
+ * addresses which are 1 byte in size would likely be able to use it.
+ *
+ * Frame Format
+ * ============
+ *
+ * ```
+ *  0                   1                   2                   3
+ *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |  Destination  |    Source     |  Payload...
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * ```
+ *
+ * | Field       | Description                     |
+ * |-------------|---------------------------------|
+ * | Destination | The layer 2 destination address |
+ * | Source      | The layer 2 source address      |
+ * | Payload     | The payload (variable size)     |
+ *
+ * Please note that the payload needs to be able to carry 6LoWPAN frames.
+ *
+ * Layer 2 Broadcast
+ * =================
+ *
+ * This adaption layer assumes that the layer 2 address `0x00` is reserved for
+ * layer 2 broadcast, which is true for CC110x and CC1200 transceivers (provided
+ * they are configured accordingly). If more users of this adaption layers are
+ * added, this behaviour might needs to be more generalized.
```

Comment:
+1 on not using a magic number.

But allowing it to be overwritten can cause trouble. Assuming another user of this adaption layer pops up that uses the code unmodified but overwrites the broadcast address e.g. to `0xff`, this device could not be used in conjunction with CC110x or CC1200 transceivers. If this case appears, I would favor to e.g. extend `cc1xxx_t` e.g. to

```C
typedef struct {
    netdev_t netdev;        /**< RIOT's interface to this driver */
    uint8_t addr;           /**< Layer 2 address of this device */
    uint8_t bcast_addr;     /**< Layer 2 address used in broadcasts  */
} cc1xxx_t;
```

(But for now I would save the one byte, as it is not yet required.)

Reply:
I know, that simultaneous usage of both wouldn't be possible then. Since this is a more hypothetical case however, I could live with that.

> (But for now I would save the one byte, as it is not yet required.)

But only on certain 8-bit platforms (however, I believe all 8-bit platforms we support align to 16-bit). However, due to the required initialization this would require some more ROM.

Reply:
The define resolves the conversation for me for now. If similar link-layers with different broadcast addresses arise we can change that.

Reply:
> If similar link-layers with different broadcast addresses arise we can change that.

Maybe by then a solution how to use the same code on simultaneously is obvious.